File: lab3/cpp/check.cpp
#include <bits/stdc++.h>
using namespace std;

long long computeHash(const string& str)
{
    int hash=0; 
    int hashfinal=0;
    
    for(int i=0; i<str.length(); i++)
    {
        int d= int(str[i]);
        
        hash= hash + ((d*((int)pow(263,i)))%1000000007);
        cout<<i<<":"<<hash<<endl;
    }
    hashfinal= hash%263;
    return abs(hashfinal);
}

int main(){
    cout<<computeHash("Rodriguez")<<endl;
    cout<<computeHash("Akshay")<<endl;
    return 0;
}

File: lab3/py/check.py
def compute_hash(string):
    term = 0
    x = 1
    calc = 0
    p = 10e9 + 7
    count = 0 

    for i in string:
        ascii = ord(i)
        term = ascii*x
        x = x*263
        term = term%p
        calc +=term
        print(count, ":", calc)
        count+=1
            
    m = 263
    calc = calc%m
    return int(calc)

print(compute_hash("Rodriguez"))
print(compute_hash("Akshay"))
